### PR TITLE
PPC: Trigger checklist item calculations after initial element added

### LINKED ISSUE #7138: PPC Trigger: Blank Story
## Description

When a user creates a new story it is blank and we should not display any messages or warnings on the pre publish checklist. 

## Implementation Suggestions 
- We should create an `isEmpty` property on story data before tackling these triggers (#7145). This will allow us to trigger the PPC when we need it while keeping the logic for what an "empty" story is out of the checklist and in the story itself. An empty story is any story with just a background color set. 

## Acceptance Criteria 
- When a user creates a new story don't trigger any PPC suggestions. No more icon next to checklist title in panel tab, no icon on publish button. Both high priority and recommended tabs of ppc are disabled. Accordion arrows are gone.
- Should also include updated unit tests 

https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7045%3A7520
Screen Shot 2021-04-07 at 3 18 56 PM

### PR DESCRIPTION
**Pull request by maxyinger**
## Context
Part of work to make prepublish checklist less annoying.

## Summary
This PR works on not calculating any checklist items until the stories been interacted with. Some of the UI updates are taken care of in this PR here: https://github.com/google/web-stories-wp/pull/7277

Where this one is only focused on altering the actual calculation of checklist items.

## User-facing changes
Checklist should now not be calculated until there's at least one element in the story.

## Testing Instructions
open a new story, see that there are no messages in the prepublish checklist. Add an element, and see that the messages now show up in the prepublish checklist.

open up an existing story with elements in it and see that right when it's open all the checklist items are present

### QA
Follow testing instructions above

- [ ] This is a non-user-facing change and requires no QA

This PR can be tested by following these steps:

1.

### UAT

- [x] UAT should use the same steps as above.

This PR can be tested by following these steps:

1.

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/google/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This PR contains automated tests (unit, integration, and/or e2e) to verify the code works as intended ([docs](https://github.com/google/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #7138

### CODE REVIEW

File: assets/src/edit-story/app/story/storyTriggers/test/useStoryTriggers.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    // Dispatching an event after a listener
+    // has cleaned up should not call that listener
+    act(() => {
+      cleanup();
+    });
+    dispatchStoryEvent = result.current[1];
+    dispatchStoryEvent(STORY_EVENTS.onInitialElementAdded);
+    expect(listenerMock).toHaveBeenCalledTimes(1);
+  });
```
Comment (samwhale) [inline assets/src/edit-story/app/story/storyTriggers/test/useStoryTriggers.js]: This is a good test

**Review by samwhale — APPROVED** [review]
noiceeee

File: assets/src/edit-story/app/prepublish/getPrepublishErrors.js [hunk 2]
```diff
@@ -26,17 +27,26 @@ import guidance from './guidance';
...
-async function getPrepublishErrors(story) {
+async function getPrepublishErrors(
+  story,
+  { types = DEFAULT_OPTS.types } = DEFAULT_OPTS
+) {
   if (!story) {
     return [];
   }
-  const checklistResult = [error, warning, guidance]
+  const checklistResult = Object.entries({ error, warning, guidance })
```
Comment (BrittanyIRL) [inline assets/src/edit-story/app/prepublish/getPrepublishErrors.js:47]: nice improvement

File: assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js [hunk 3]
```diff
@@ -46,6 +46,8 @@ describe('Pre-publish checklist select offending elements onClick', () => {
     const { checklistTab } = fixture.editor.inspector;
     await fixture.events.click(checklistTab);
     await waitFor(() => fixture.editor.inspector.checklistPanel);
+    // just add a small wait for the checklist to compute it's items
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js]: (nit :) )
```suggestion
    // just add a small wait for the checklist to compute its items
```

File: assets/src/edit-story/components/inspector/prepublish/checklistTab.js [hunk 4]
```diff
@@ -188,60 +188,62 @@ const ChecklistTab = ({ checklist, currentCheckpoint }) => {
...
+    !hasRecommendedItems &&
+    !isHighPriorityDisabledState;
 
-  if (!showHighPriorityItems && !showRecommendedItems) {
-    return <EmptyChecklist />;
-  }
+  const isRecommendedDisabled =
+    isRecommendedDisabledState || !hasRecommendedItems;
+  const isHighPriorityDisabled =
+    isHighPriorityDisabledState || !hasHighPriorityItems;
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/inspector/prepublish/checklistTab.js:233]: i like your updates to this flow!

**Review by BrittanyIRL — APPROVED** [review]
Awesome!! Looks great. Thanks for all the refactor too.

File: assets/src/edit-story/components/inspector/prepublish/test/prepublishChecklistProvider.js [hunk 5]
```diff
@@ -62,29 +63,32 @@ const generateStoryPages = (pageCount) => {
...
   const wrapper = ({ children }) => (
     <StoryContext.Provider value={storyContext}>
-      <LayoutProvider value={layoutContext}>
-        <PrepublishChecklistProvider>{children}</PrepublishChecklistProvider>
-      </LayoutProvider>
+      <StoryTriggersProvider story={fullStory}>
+        <LayoutProvider value={layoutContext}>
+          <PrepublishChecklistProvider>{children}</PrepublishChecklistProvider>
+        </LayoutProvider>
+      </StoryTriggersProvider>
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/inspector/prepublish/test/prepublishChecklistProvider.js:91]: the below test is now actually working so it should actually expect `UNAVAILABLE` not `ALL`

File: assets/src/edit-story/components/inspector/prepublish/prepublishChecklistProvider.js [hunk 6]
```diff
@@ -33,6 +34,10 @@ import {
 } from './prepublishCheckpointState';
 
 function PrepublishChecklistProvider({ children }) {
+  const [checkpointState, dispatch] = useReducer(
+    checkpointReducer,
+    PPC_CHECKPOINT_STATE.UNAVAILABLE
```
Comment (samwhale) [inline assets/src/edit-story/components/inspector/prepublish/prepublishChecklistProvider.js:39]: 👍

**Review by samwhale — APPROVED** [review]
like the updates

**Comment by BrittanyIRL** [comment]
@googlebot I consent.

File: assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js [hunk 7]
```diff
@@ -62,6 +64,25 @@ describe('Pre-publish checklist select offending elements onClick', () => {
...
+
+  async function enableHighPriorityMessagesWith5Pages() {
+    await duplicatePage();
+    await duplicatePage();
+    await duplicatePage();
+    await duplicatePage();
+    await duplicatePage();
+    await openPrepublishPanel();
+    await waitFor(
+      () => !fixture.editor.inspector.checklistPanel.highPriority.disabled
```
Comment (BrittanyIRL) [inline assets/src/edit-story/components/inspector/karma/prepublishSelect.karma.js:82]: this was smart!

**Review by BrittanyIRL — APPROVED** [review]
Your test fixes are great and i love the addition to track panel expansion in the checklist. Thanks for adding these!

**Pull request merged**